### PR TITLE
bitmart: updated the fetchTickers spot endpoint

### PR DESCRIPTION
**Pull request by Dan-krm**
Updated the spot endpoint for fetchTickers and fetchTicker:
```
node examples/js/cli bitmart fetchTickers '["BTC/USDT","ETH/USDT"]'

bitmart.fetchTickers (BTC/USDT,ETH/USDT)
2024-05-31T04:38:48.799Z iteration 0 passed in 1006 ms

{
  'BTC/USDT': {
    symbol: 'BTC/USDT',
    timestamp: 1717130327808,
    datetime: '2024-05-31T04:38:47.808Z',
    high: 69499.99,
    low: 67132.4,
    bid: 68508.59,
    bidVolume: 0.00627,
    ask: 68508.61,
    askVolume: 0.01234,
    vwap: 68354.4118269248,
    open: 68303.27,
    close: 68508.6,
    last: 68508.6,
    change: 0.00301,
    percentage: 0.301,
    average: 68405.935,
    baseVolume: 10533.01836,
    quoteVolume: 719978274.76,
    info: [
      'BTC_USDT',      '68508.60',
      '10533.01836',   '719978274.76',
      '68303.27',      '69499.99',
      '67132.40',      '0.00301',
      '68508.59',      '0.00627',
      '68508.61',      '0.01234',
      '1717130327808'
    ]
  },
  'ETH/USDT': {
    symbol: 'ETH/USDT',
    timestamp: 1717130327783,
    datetime: '2024-05-31T04:38:47.783Z',
    high: 3825.35,
    low: 3702.6,
    bid: 3759.99,
    bidVolume: 0.00532,
    ask: 3760,
    askVolume: 0.02055,
    vwap: 3758.4100324692363,
    open: 3794.82,
    close: 3760,
    last: 3760,
    change: -0.00918,
    percentage: 0.918,
    average: 3777.41,
    baseVolume: 113844.1009,
    quoteVolume: 427872810.96,
    info: [
      'ETH_USDT',      '3760.00',
      '113844.10090',  '427872810.96',
      '3794.82',       '3825.35',
      '3702.60',       '-0.00918',
      '3759.99',       '0.00532',
      '3760',          '0.02055',
      '1717130327783'
    ]
  }
}
```
```
bitmart.fetchTicker (BTC/USDT)
2024-05-31T05:12:04.737Z iteration 0 passed in 475 ms

{
  symbol: 'BTC/USDT',
  timestamp: 1717132325396,
  datetime: '2024-05-31T05:12:05.396Z',
  high: 69499.99,
  low: 67132.4,
  bid: 68479.98,
  bidVolume: 0.05972,
  ask: 68480,
  askVolume: 0.00058,
  vwap: 68360.17033443128,
  open: 68078.42,
  close: 68479.99,
  last: 68479.99,
  change: 0.0059,
  percentage: 0.59,
  average: 68279.205,
  baseVolume: 10530.35952,
  quoteVolume: 719857170.47,
  info: {
    symbol: 'BTC_USDT',
    last: '68479.99',
    v_24h: '10530.35952',
    qv_24h: '719857170.47',
    open_24h: '68078.42',
    high_24h: '69499.99',
    low_24h: '67132.40',
    fluctuation: '0.00590',
    bid_px: '68479.98',
    bid_sz: '0.05972',
    ask_px: '68480',
    ask_sz: '0.00058',
    ts: '1717132325396'
  }
}
```

### CODE REVIEW

**Comment by carlosmiei** [comment]
@Dan-krm can you rebase this pls?

File: ts/src/bitmart.ts [hunk 1]
```diff
@@ -1184,21 +1201,22 @@ export default class bitmart extends Exchange {
         //          "legal_coin_price":"0.1302699"
         //      }
         //
-        let timestamp = this.safeInteger (ticker, 'timestamp');
+        let timestamp = this.safeIntegerN (ticker, [ 'timestamp', 'ts', 12 ]);
```
Comment (carlosmiei) [inline ts/src/bitmart.ts:1204]: Hum, this is tricky. The safeMethods won't work properly if the ticker can either be a list or a direct.

Comment (Dan-krm) [reply ts/src/bitmart.ts:1204]: I could separate the list values into a dictionary in `fetchTicker` and `fetchTickers` then parse the new dictionary if that works

Comment (carlosmiei) [reply ts/src/bitmart.ts:1204]: @Dan-krm you don't need to convert the list, you can do something like

```
let high = this.safeString(ticker, 'h') // read from dict
if (high == undefined) {
   high = this.safeString(ticker, 0) // read from list
}
```
besides that parseTicker/parseTickers expect a dict and not a list, so we might need to create a wrapper around the list, please check how we do it in bitfinex2

Comment (Dan-krm) [reply ts/src/bitmart.ts:1204]: Okay I'm working on it

Comment (Dan-krm) [reply ts/src/bitmart.ts]: I created a dict wrapper for spot fetchTickers and separated reading the values from a dict or list in parseTicker

**Comment by carlosmiei** [comment]
@Dan-krm can you check the build pls

**Pull request merged**